src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1105,6 +1134,12 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
 
                 addressTransactions.Add(newTransaction);
                 this.AddInputKeysLookupLocked(newTransaction);
+
+                if (block == null)
```

Comment:
Ok so here, if the transaction is by itself, you add the inputs in the lookup table against the newly created TransactionData...